src/Workleap.OpenApi.MSBuild/Spectral/SpectralRunner.cs

```diff
@@ -66,10 +68,35 @@ public async Task RunSpectralAsync(IReadOnlyCollection<string> openApiDocumentPa
             CiReportRenderer.AttachReportToBuild(spectralReportPath);
             this._loggerWrapper.LogMessage("\n ****************************************************************", MessageImportance.High);
         }
-        
+
         this._diffCalculator.SaveCurrentExecutionChecksum(spectralRulesetPath, openApiDocumentPaths);
     }
-    
+
+    // Display previous spectral report and log warning if there are any previous spectral errors or warnings
+    private void DisplayPreviousSpectralReport(string documentPath)
+    {
+        var pattern = @"\d+ problem(|s) \((\d+) error(|s), (\d+) warning(|s), \d+ info(|s), \d+ hint(|s)\)";
```

Comment:
nit: a couple of things:

1. I usually prefer to declare Regex instances as static readonly fields as they can have a better name reflecting what it does than just `pattern` and `regex`.
2. I also like to put a comment of the kind of sentence it's supposed to parse on top of the regex
3. We don't need groups to handle optional s characters, we can use `s?` instead of `(|s)`
4. I like to use [named capture groups](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) to provide more context when accessing the group value later, like:

```regex
\d+ problems? \((?<errors>\d+) errors?, (?<warnings>\d+) warnings?, \d+ infos?, \d+ hints?\)
```

... so you should be able to access the groups like this:

```
var errors = int.Parse(match.Groups["errors"].Value);
var warnings = int.Parse(match.Groups["warnings"].Value);
```

Reply:
Ooh, thanks for sharing, added those changes